src/Microsoft.SqlTools.ServiceLayer/DacFx/GenerateTSqlModelOperation.cs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Diagnostics;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.SqlServer.Dac.Model;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent creating a dacfx model
+    /// </summary>
+    class GenerateTSqlModelOperation
+    {
+        public GenerateTSqlModelParams Parameters { get; }
+
+        public GenerateTSqlModelOperation(GenerateTSqlModelParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        /// <summary>
+        /// Generate model from sql files, if no sql files are passed in then it generates an empty model.
+        /// </summary>
+        public TSqlModel GenerateTSqlModel()
+        {
+            try
```

Comment:
I don't see any tests. Are you planning to add any in this PR?

Reply:
I'll add some tests